datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -2434,6 +2434,21 @@ mod tests {
         // single word
         assert_change(regex_match(col("c1"), lit("foo")), like(col("c1"), "%foo%"));
 
+        // regular expressions that match an exact literal
+        assert_change(regex_match(col("c1"), lit("^$")), col("c1").eq(lit("")));
+        assert_change(
+            regex_not_match(col("c1"), lit("^$")),
+            col("c1").not_eq(lit("")),
+        );
+        assert_change(
+            regex_match(col("c1"), lit("^foo$")),
+            col("c1").eq(lit("foo")),
+        );
+        assert_change(
+            regex_not_match(col("c1"), lit("^foo$")),
+            col("c1").not_eq(lit("foo")),
+        );
```

Comment:
Could we have another "unsupported" test here for (use `assert_no_change`, see other cases in this test):

- `^foo|bar$` (OR within anchor)
- `^(foo)(bar)$` (I think this is a concat or more than 3 elements)
- `^` and `$` (single element)
- `$^` and `$foo^` (anchors flipped)

Reply:
Done, also added a test that combines anchored regexes with OR-chains, since `lower_simple` is called recursively for that case.